src/mbgl/annotation/annotation_tile.hpp

```diff
@@ -45,7 +45,10 @@ class AnnotationTileLayer : public GeometryTileLayer {
     AnnotationTileLayer(std::string);
 
     std::size_t featureCount() const override { return features.size(); }
-    std::unique_ptr<GeometryTileFeature> getFeature(std::size_t i) const override { return std::make_unique<AnnotationTileFeature>(features[i]); }
+    std::unique_ptr<GeometryTileFeature> getFeature(std::size_t i) const override {
+        assert(i < features.size());
+        return std::make_unique<AnnotationTileFeature>(features[i]);
```

Comment:
Check. I've changed this, but since `vector::at` throws an exception, which will still crash the app, I'm wondering if we shouldn't catch, log an error and continue in these cases. At least in release builds.